client/AI/AIClientApp.cpp

```diff
@@ -141,7 +141,10 @@ void AIClientApp::Run() {
                                                  boost::uuids::nil_uuid()));
 
         // Start parsing content
-        StartBackgroundParsing();
+        std::promise<void> barrier;
+        std::future<void> barrier_future = barrier.get_future();
```

Comment:
Could we move this barrier code inside the StartBackgroundParsing and return barrier_future from there. This pattern repeats a couple of times.

Result might look like:
```cpp

barrier_future = StartBackgroundParsing()
barrier_future.wait()
```

Reply:
No, I need to wait for barrier future outside of background parsing scope and thread in case of the UI client.